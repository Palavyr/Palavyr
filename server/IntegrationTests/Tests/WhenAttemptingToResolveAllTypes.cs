using System;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using IntegrationTests.AppFactory.AutofacWebApplicationFactory;
using IntegrationTests.AppFactory.IntegrationTestFixtures;
using Test.Common.ApprovalTests;
using Xunit;
using Xunit.Abstractions;

namespace IntegrationTests.Tests
{
    public class WhenAttemptingToResolveAllTypes : IntegrationTest
    {
        private readonly ITestOutputHelper testOutputHelper;

        [Fact]
        public void AllTypesAreResolvedSuccessfully()
        {
            var allTypes = AppDomain
                .CurrentDomain
                .GetAssemblies()
                .Where(x => x.FullName!.Contains("Palavyr"))
                .SelectMany(x => x.GetTypes())
                .Where(x => x.IsInterface)
                .ToList();

            var stringBuilder = new StringBuilder();

            foreach (var t in allTypes)
            {
                try
                {
                    var instance = Container.GetService(t);
                    if (instance is null)
                        stringBuilder.Append($"{t.Name}\n");
                }
                catch (Exception)
                {
                    stringBuilder.Append($"{t.Name}\n");
                }
            }

            this.PalavyrAssent(stringBuilder.ToString());
        }

        [Fact]
        public void AllHandlersAreResolved()
        {
            // var containerBuilder = new ContainerBuilder();
            // var config = TestConfiguration.GetTestConfiguration();
            //
            // Startup.ContainerSetup(containerBuilder, config);
            // var iContainer = containerBuilder.Build();
            // var data = new AutofacRegistrations(iContainer.ComponentRegistry);
            //
            // Assert.Throws<ConventionFailedException>(() => Convention.Is(new CanResolveAllRegisteredServices(iContainer), data));
        }


        public WhenAttemptingToResolveAllTypes(ITestOutputHelper testOutputHelper, ServerFactory factory)
            : base(testOutputHelper, factory)
        {
            this.testOutputHelper = testOutputHelper;
        }

        public override async Task DisposeAsync()
        {
            await Task.CompletedTask;
            WebHostFactory.Dispose();
        }
    }
}