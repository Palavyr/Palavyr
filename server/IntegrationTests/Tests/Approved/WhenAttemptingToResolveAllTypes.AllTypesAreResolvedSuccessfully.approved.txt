IGetEmailSent
IDynamicTableController`2
IForbiddenEntityStore
IEntityStore`1
IHaveTempDirectory
IHaveLocalFileName
IHaveTemporaryPath
IHaveBeenDownloadedFromCloudToLocal
IPricingStrategyTableCompiler
IPricingStrategyTableCommandExecutor`3
IResponseRetriever`1
IFileAssetLinker`1
IResolveCloudKeys
IPricingStrategyTableRowResource
IAccount
IEntity
IHaveAccountId
IHaveRange
IId
IMultiItem
IMultiRowEntity
IOrderableThreshold
IOrderedRow
IOrderedTable
IRecord
ISingleRowEntity
IStaticTable
IStaticTableRow
ITable
IPricingStrategyTable`1
IHaveAPrettyNameAndTableType
ICreatePricingStrategyTemplate`1
IMapToNew`2
IMapToPreExisting`2
IProcessStripeInvoicePaymentFailedHandler
IPricingStrategyTemplateCreator`1
IPalavyrClient
